app/src/main/kotlin/com/backbase/golden_sample_app/user/ProfileRepository.kt

```diff
@@ -0,0 +1,35 @@
+package com.backbase.golden_sample_app.user
+
+import com.backbase.android.client.usermanagerclient2.api.UserProfileManagementApi
+import com.backbase.android.client.usermanagerclient2.model.UserProfile
+import com.backbase.android.secure.storage.SecureStorage
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.withContext
+
+class ProfileRepository(
+    private val remoteSource: UserProfileManagementApi,
+    private val localSource: SecureStorage,
```

Comment:
This is not really abstracted. Also, is it needed to store this locally? I believe in memory is good enough.

Reply:
Local Storage in my opinion is not needed, we can just store in memory. 

About remoteSource and localSource. Not sure if we need to apply that naming when there is no abstraction: it requires UserProfileManagementApi and SecureStorage